### PR TITLE
add delete button and popup

### PR DESCRIPTION
**Pull request by Mengzhen1992**
please check this feature: 

1. add delete button
2. add popup

### CODE REVIEW

File: components/Popup.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        <Button onClick={handleDeleteTrue}>Yes, delete it</Button>
+        <Button onClick={handleDeleteFalse}>Cancel</Button>
+      </PopupContainer>
+    </Overlay>
+  );
+}
+
+const Overlay = styled.div`
+  position: fixed;
+  height: 100%;
```
Comment (ThomasF85) [inline components/Popup.js:19]: Are you sure you want the overlay to be 100% in width and height, not 100vh and 100vw?

File: components/Popup.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  right: 1.5rem;
+  top: 25%;
+  bottom: 40%;
+  padding: 2rem;
+  background: rgba(255, 255, 255, 0.8);
+  box-shadow: 0px 4px 20px rgba(0, 0, 0, 0.1);
+  border-radius: 8px;
+`;
+
+const Text = styled.h2`
```
Comment (ThomasF85) [inline components/Popup.js]: The name Text does not say much. It is a heading you have here.

File: pages/index.js [hunk 3]
```diff
@@ -1,11 +1,32 @@
...
+import { useState } from "react";
 
 export default function Home({ tasks, setTasks, getCurrentDate }) {
+  const [popup, setPopup] = useState({ show: false, id: null });
+  // this will show the confirmation box
+  function handleDelete(id) {
+    setPopup({ show: true, id });
+  }
+  // this will perform the deletion and hide the confirmation Box
+  function handleDeleteTrue() {
```
Comment (ThomasF85) [inline pages/index.js]: The naming of this function is unclear. You should name your functions like onDelete, onCancelDelete,... Then the name of the function alone is telling what the function does.

**Review by ThomasF85 — APPROVED** [review]
Good job! I Have some minor requests. Please have a look at them before merging.

File: components/Popup.js (hunk 1 above)
Comment (Mengzhen1992) [reply components/Popup.js:19]: @ThomasF85 i have tried both, when the overlay is 100vh and 100vw in height and width, if the content becomes longer than my background and i click on the task item at the bottom, the overlay can not cover the background and looks strange, like this: 
Screenshot 2022-10-25 at 15 57 10

File: components/Popup.js (hunk 2 above)
Comment (Mengzhen1992) [reply components/Popup.js]: @ThomasF85 good suggestion! Thanks 👍

**Pull request merged**